gym_wrapper/base_gym_env.py

```diff
@@ -0,0 +1,195 @@
+        if self.num_game_variables > 0:
+            spaces["gamevariables"] = gym.spaces.Box(
+                np.finfo(np.float32).min,
+                np.finfo(np.float32).max,
+                (self.num_game_variables,),
+                dtype=np.float32
+            )
+
+        self.observation_space = gym.spaces.Dict(spaces)
+
+    def step(self, action):
+        assert self.action_space.contains(action), f"{action!r} ({type(action)}) invalid"
+        assert self.state is not None, "Call `reset` before using `step` method."
+
+        # convert action to vizdoom action space (one hot)
+        act = [0 for _ in range(self.action_space.n)]
+        act[action] = 1
+
+        reward = self.game.make_action(act, self.frame_skip)
+        self.state = self.game.get_state()
+        done = self.game.is_episode_finished()
+
+        return self.__collect_observations(), reward, done, {}
+
+    def reset(
+        self,
+        *,
+        seed: Optional[int] = None,
+        return_info: bool = False,
+        options: Optional[dict] = None,
+    ):
+        if seed is not None:
+            self.game.set_seed(seed)
+        self.game.new_episode()
+        self.state = self.game.get_state()
+
+        if not return_info:
+            return self.__collect_observations()
+        else:
+            return self.__collect_observations(), {}
+
+    def __collect_observations(self):
+        observation = {}
+        if self.state is not None:
+            # Check for GRAY8 (no channel dim)
+            screen_buffer = self.state.screen_buffer
+            if screen_buffer.ndim == 2:
+                # Add extra dimension for channel
+                screen_buffer = screen_buffer[None]
+            observation["rgb"] = np.transpose(screen_buffer, (1, 2, 0))
+            if self.depth:
+                observation["depth"] = self.state.depth_buffer
+            if self.labels:
+                observation["labels"] = self.state.labels_buffer
+            if self.num_game_variables > 0:
+                observation["gamevariables"] = self.state.game_variables.astype(np.float32)
+        else:
+            # there is no state in the terminal step, so a zero observation is returned instead
+            for space_key, space_item in self.observation_space.spaces.items():
+                observation[space_key] = np.zeros(space_item.shape, dtype=space_item.dtype)
+
+        return observation
+
+    def render(self, mode="human"):
+        game_state = self.game.get_state()
+        if game_state is None:
+            img = np.zeros(
+                (
+                    self.game.get_screen_channels(),
+                    self.game.get_screen_height(),
+                    self.game.get_screen_width(),
+                ),
+                dtype=np.uint8,
+            )
+        else:
+            img = game_state.screen_buffer
+        img = np.transpose(img, [2, 1, 0])
```

Comment:
As before, this only works for CRCGCB format. And produce error for GRAY8 format:
```
Traceback (most recent call last):
  File "gym_wrapper.py", line 29, in <module>
    env.render()
  File "<python dir>/python3.8/site-packages/gym/core.py", line 286, in render
    return self.env.render(mode, **kwargs)
  File "<python dir>/python3.8/site-packages/vizdoom/gym_wrapper/base_gym_env.py", line 171, in render
    img = np.transpose(img, [2, 1, 0])
  File "<__array_function__ internals>", line 180, in transpose
  File "<python dir>/python3.8/site-packages/numpy/core/fromnumeric.py", line 660, in transpose
    return _wrapfunc(a, 'transpose', axes)
  File "<python dir>/python3.8/site-packages/numpy/core/fromnumeric.py", line 57, in _wrapfunc
    return bound(*args, **kwds)
ValueError: axes don't match array
```